### PR TITLE
Write image digest metadata file.

### PR DESCRIPTION
**Pull request by fxb**

### CODE REVIEW

**Comment by mattnworb** [comment]
IIRC the image digest is only set after the image is pushed to a registry

**Comment by dflemstr** [comment]
@fxb will you address the test failures?

File: plugin/src/it/basic-lowercase-dockerfile/verify.groovy [hunk 1]
```diff
@@ -19,3 +19,6 @@
  */
 File imageIdFile = new File(basedir, "target/docker/image-id")
 assert imageIdFile.isFile()
+
+File imageDigestFile = new File(basedir, "target/docker/image-digest")
+assert imageDigestFile.isFile()
```
Comment (dflemstr) [inline plugin/src/it/basic-lowercase-dockerfile/verify.groovy]: Would prefer if you create a new test that only tests this functionality, all the tests don't need to test it as well.

File: plugin/src/main/java/com/spotify/plugin/dockerfile/BuildMojo.java [hunk 2]
```diff
@@ -73,6 +73,20 @@
   @Parameter(property = "dockerfile.build.pullNewerImage", defaultValue = "true")
   private boolean pullNewerImage;
 
+  /**
+   * Return value of buildImage method, holding image id and digest.
+   */
+  private static class ImageInfo {
+
```
Comment (dflemstr) [inline plugin/src/main/java/com/spotify/plugin/dockerfile/BuildMojo.java]: Consider using `@AutoValue`

File: plugin/src/main/java/com/spotify/plugin/dockerfile/BuildMojo.java [hunk 3]
```diff
@@ -73,6 +73,20 @@
...
+    public ImageInfo(String imageId, String imageDigest) {
+      this.imageId = imageId;
+      this.imageDigest = imageDigest;
+    }
+
+    public String imageId;
```
Comment (dflemstr) [inline plugin/src/main/java/com/spotify/plugin/dockerfile/BuildMojo.java]: Either mark these `@Nullable` or make them `Optional<String>`

File: plugin/src/it/multi-module/verify.groovy [hunk 4]
```diff
@@ -17,9 +17,12 @@
  * limitations under the License.
  * -\-\-
  */
-File imageIdFile = new File(basedir, "b/target/docker-info-deps/META-INF/docker/com.spotify.it/a/image-id")
+File imageIdFile = new File(basedir, ""b/target/docker-info-deps/META-INF/docker/com.spotify.it/a/image-id")
```
Comment (dflemstr) [inline plugin/src/it/multi-module/verify.groovy]: Syntax error?

**Pull request closed without merging**